test/e2e/installplan_e2e_test.go

```diff
@@ -2998,6 +3000,171 @@ var _ = Describe("Install Plan", func() {
+			ns         *corev1.Namespace
+			catsrcName string
+			crc        versioned.Interface
+			ip         *operatorsv1alpha1.InstallPlan
+		)
+		BeforeEach(func() {
+			crc = newCRClient()
+			ns = &corev1.Namespace{}
+			ns.SetName(genName("ns-"))
+			Expect(ctx.Ctx().Client().Create(context.Background(), ns)).To(Succeed())
+
+			// Create the single (kiali) bundle catalog source
+			catsrc := &operatorsv1alpha1.CatalogSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      genName("kiali-"),
+					Namespace: ns.GetName(),
+					Labels:    map[string]string{"olm.catalogSource": "kaili-catalog"},
+				},
+				Spec: operatorsv1alpha1.CatalogSourceSpec{
+					Image:      "quay.io/olmtest/single-bundle-index:1.0.0",
+					SourceType: operatorsv1alpha1.SourceTypeGrpc,
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), catsrc)).To(Succeed())
+
+			// Wait for the CatalogSource to be ready
+			catsrc, err := fetchCatalogSourceOnStatus(crc, catsrc.GetName(), catsrc.GetNamespace(), catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+
+			catsrcName = catsrc.GetName()
+
+			// Create the OperatorGroup
+			og := &operatorsv1.OperatorGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "og",
+					Namespace: ns.GetName(),
+				},
+				Spec: operatorsv1.OperatorGroupSpec{
+					TargetNamespaces: []string{ns.GetName()},
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), og)).To(Succeed())
+
+			now := metav1.Now()
+			ip = &operatorsv1alpha1.InstallPlan{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ip",
+					Namespace: ns.GetName(),
+				},
+				Spec: operatorsv1alpha1.InstallPlanSpec{
+					ClusterServiceVersionNames: []string{"foobar"},
+					Approval:                   v1alpha1.ApprovalAutomatic,
+					Approved:                   true,
+				},
+				Status: operatorsv1alpha1.InstallPlanStatus{
+					Phase:          operatorsv1alpha1.InstallPlanPhaseInstalling,
+					CatalogSources: []string{},
+					BundleLookups: []operatorsv1alpha1.BundleLookup{
+						{
+							Identifier: "foobar.v0.0.1",
+							CatalogSourceRef: &corev1.ObjectReference{
+								Namespace: ns.GetName(),
+								Name:      catsrcName,
+							},
+							Conditions: []operatorsv1alpha1.BundleLookupCondition{
+								{
+									Type:               operatorsv1alpha1.BundleLookupPending,
+									Status:             corev1.ConditionTrue,
+									Reason:             "JobIncomplete",
+									Message:            "unpack job not completed",
+									LastTransitionTime: &now,
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		AfterEach(func() {
+			Expect(ctx.Ctx().Client().Delete(context.Background(), ns)).To(Succeed())
+		})
+
+		It("should show an error on the bundlelookup condition for a non-existent bundle image", func() {
+			// Create an InstallPlan status.bundleLookups.Path specified for a non-existent bundle image
+			ip.Status.BundleLookups[0].Path = "quay.io/foo/bar:v0.0.1"
```

Comment:
This could break if someone publishes to quay.io/foo/bar. Does the test work if this points to something that isn't a registry, like `localhost:0/does/not:exist`?

Reply:
It does. Will update this.